### ISSUE TITLE
Observable.take() on a BehaviorSubject swallows "fatal" errors.

### ISSUE THREAD

**Issue by mikebaum** [ReactiveX/RxJava#3095]
It seems that if your subscriber throws a "fatal" exception when subscribing to a BehaviorSubject chained with just, the fatal there error is swallowed and nothing is logged. See the following example:

```
    public static void main( String[] args )
    {
        BehaviorSubject<String> strings = BehaviorSubject.create( "First" );
        
        RxJavaPlugins.getInstance().registerErrorHandler( new RxJavaErrorHandler()
        {
            public void handleError( Throwable throwable )
            {
                System.err.println( throwable );
            };
        } );
        
        strings.take( 1 ).subscribe( string -> { throw new NoClassDefFoundError(); },
                                     System.err::println );
    }
```

If the error thrown from the subscribe function is anything but a "fatal" error as determined by `Exceptions.throwIfFatal(Throwable)` an error is properly propagated.

**Comment by akarnokd** [comment]
This is a strange situation: take sets its completed flag before the emission and when the exception is bounced back, it is not propagated any further. Looks like a bug.

**Comment by mikebaum** [comment]
@akarnokd Thanks for the quick bug fix!

**Comment by akarnokd** [comment]
The bug is fixed and is now part of 1.0.14. Thanks for reporting.
